packages/access-client/src/agent-data.js

```diff
@@ -111,12 +111,15 @@ export class AgentData {
 
   /**
    * @deprecated
-   * @param {import('@ucanto/interface').DID} did
+   * @param {import('@ucanto/interface').DID<'key'>} did
    * @param {import('./types').SpaceMeta} meta
    * @param {import('@ucanto/interface').Delegation} [proof]
    */
   async addSpace(did, meta, proof) {
     this.spaces.set(did, meta)
+    if (!this.currentSpace) {
+      this.currentSpace = did
```

Comment:
Side effects make me nervous. I'd feel better about it if the agent class called setCurrent after adding a space(s)...at this level there's no choice for anyone using the data class.

Please if you decide not to move then can you at least add a comment to the method in the agent class to say that it will set current space if one is not set? 🙏